modal/_container_entrypoint.py

```diff
@@ -699,7 +699,7 @@ def main(container_args: api_pb2.ContainerArguments, client: Client):
 
     _client: _Client = synchronizer._translate_in(client)  # TODO(erikbern): ugly
 
-    with container_io_manager.heartbeats(), UserCodeEventLoop() as event_loop:
+    with container_io_manager.heartbeats(function_def.is_checkpointing_function), UserCodeEventLoop() as event_loop:
```

Comment:
Are we just disabling heartbeats if we are checkpointing? If so, isn't that dangerous? i.e. functions may execute long-running work and missing heartbeats may kill it

Reply:
We disable heartbeats only during the checkpoint phase. They are re-enabled after the container restores. So in any case, heartbeats will be running whenever long-running user code is.